### PR TITLE
Add is_default to connections list

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Adds `is_default` column to `snow connection list`.

### CODE REVIEW

File: tests/plugin/test_broken_plugin.py [hunk 1]
```diff
@@ -1,25 +1,12 @@
...
-def test_broken_command_path_plugin(runner, test_root_path, _install_plugin, caplog):
+def test_broken_command_path_plugin(runner, test_root_path, _install_plugin, snapshot):
     config_path = test_root_path / "test_data" / "configs" / "broken_plugin_config.toml"
 
     result = runner.invoke(["--config-file", config_path, "connection", "list"])
 
-    assert (
-        caplog.messages[0]
-        == "Cannot register plugin [broken_plugin]: Invalid command path [snow broken run]. Command group [broken] does not exist."
-    )
```
Comment (sfc-gh-astus) [inline tests/plugin/test_broken_plugin.py:14]: We can't remove this check, it is requirement from RA

File: tests/plugin/test_failing_plugin.py [hunk 2]
```diff
@@ -1,27 +1,14 @@
...
     config_path = (
         test_root_path / "test_data" / "configs" / "failing_plugin_config.toml"
     )
 
     result = runner.invoke(["--config-file", config_path, "connection", "list"])
 
-    assert (
-        caplog.messages[0]
-        == "Cannot register plugin [failing_plugin]: Some error in plugin"
-    )
```
Comment (sfc-gh-astus) [inline tests/plugin/test_failing_plugin.py:16]: Same here

File: tests/plugin/test_override_by_external_plugins.py [hunk 3]
```diff
@@ -1,32 +1,18 @@
...
     config_path = (
         test_root_path / "test_data" / "configs" / "override_plugin_config.toml"
     )
 
     result = runner.invoke(["--config-file", config_path, "connection", "list"])
 
-    assert (
-        caplog.messages[0]
-        == "Cannot register plugin [override]: Cannot add command [snow connection list] because it already exists."
-    )
```
Comment (sfc-gh-astus) [inline tests/plugin/test_override_by_external_plugins.py:16]: And here

File: tests/plugin/test_override_by_external_plugins.py [hunk 4]
```diff
@@ -37,15 +23,7 @@ def test_disabled_plugin_is_not_executed(
 
     result = runner.invoke(["--config-file", config_path, "connection", "list"])
 
-    assert len(caplog.messages) == 0
```
Comment (sfc-gh-astus) [inline tests/plugin/test_override_by_external_plugins.py:40]: Same

Comment (sfc-gh-turbaszek) [reply tests/plugin/test_override_by_external_plugins.py:40]: So, we use weird text checking only because we don't trust snapshots?

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**